uni/lib/controller/parsers/parser_courses.dart

```diff
@@ -1,38 +1,80 @@
-import 'dart:collection';
-
 import 'package:html/parser.dart' show parse;
 import 'package:http/http.dart' as http;
+import 'package:uni/model/entities/course.dart';
 
-Map<String, String> parseMultipleCourses(List<http.Response> responses) {
-  final Map<String, String> coursesStates = HashMap();
+List<Course> parseMultipleCourses(List<http.Response> responses) {
+  final List<Course> courses = [];
   for (var response in responses) {
-    final map = parseCourses(response);
-    coursesStates.addAll(map);
+    final c = parseCourses(response);
+    courses.addAll(c);
   }
-  return coursesStates;
+  return courses;
 }
 
-/// Extracts a map containing information about the user's courses from an HTTP
-/// [response].
-///
-/// *Note:*
-/// * a key in this map is the name of a course
-/// * a value in this map is the state of the corresponding course
-Map<String, String> parseCourses(http.Response response) {
+List<Course> parseCourses(http.Response response) {
   final document = parse(response.body);
+  final List<Course> courses = [];
 
-  final Map<String, String> coursesStates = HashMap();
+  String? faculty = response.request?.url.toString();
+  faculty =
+      faculty?.substring(faculty.indexOf('up.pt/')).replaceFirst('up.pt/', '');
+  faculty = faculty?.substring(0, faculty.indexOf('/')).trim();
 
-  final courses =
+  final currentCourses =
       document.querySelectorAll('.estudantes-caixa-lista-cursos > div');
+  for (int i = 0; i < currentCourses.length; i++) {
+    final div = currentCourses[i];
+    final courseName =
+        div.querySelector('.estudante-lista-curso-nome > a')?.text;
+    final courseUrl = div
+        .querySelector('.estudante-lista-curso-nome > a')
+        ?.attributes['href'];
+    var courseId =
+        courseUrl?.replaceFirst('cur_geral.cur_view?pv_curso_id=', '').trim();
```

Comment:
Regex maybe could improve the way to handle the arguments, but i don't know much about it, and neither does the team. 
Maybe creating a util function to extract the parameters into a Map would be a better way to handle the url parameters

Reply:
Sure